cirq-google/cirq_google/devices/google_device.py

```diff
@@ -0,0 +1,226 @@
+# Copyright 2022 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Device object representing Google devices."""
+
+from typing import Any, Set, Tuple, cast
+import cirq
+from cirq_google.api import v2
+
+
+@cirq.value_equality
+class GoogleDevice(cirq.Device):
```

Comment:
Might want to call it something like `UniformGoogleDevice` or `GoogleGridDevice` just to convey the message that this type identifies and works for devices that have uniform gates in all directions etc. Since we are surfacing a GridDeviceMetadata here we want to ensure that the user knows they won't get a device that doesn't have the same gates and connectivity everywhere.

Reply:
Sounds good. How about just `GridDevice`? I'm cautious with "uniform" because for example some 2-qubit gates may be missing. The "Google" descriptor is implicit from the package name and was there only because I couldn't think of a better descriptor :)